File: PronicNumber.c
#include <stdio.h>

int isPronic(int num) {
    if (num < 0) return 0;

    for (int i = 0; i * (i + 1) <= num; i++) {
        if (i * (i + 1) == num)
            return 1;
    }

    return 0;
}

void main() {
    int num;

    printf("Enter a number: ");
    scanf("%d", &num);

    if (isPronic(num))
        printf("%d is a Pronic number", num);
    else
        printf("%d is not a Pronic number", num);
}

File: PronicNumber.java
import java.util.Scanner;

public class PronicNumber {

    public static boolean isPronic(int num) {
        if (num < 0) return false;

        for (int i = 0; i * (i + 1) <= num; i++) {
            if (i * (i + 1) == num)
                return true;
        }

        return false;
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = sc.nextInt();

        if (isPronic(num))
            System.out.println(num + " is a Pronic number");
        else
            System.out.println(num + " is not a Pronic number");

        sc.close();
    }
}